components/Newsfeed/Newsfeed.tsx

```diff
@@ -0,0 +1,254 @@
+
+export default function Newsfeed() {
+  const isMobile = useMediaQuery("(max-width: 768px)")
+
+  const { user } = useAuth()
+  const uid = user?.uid
+  const { result: profile, loading } = usePublicProfile(uid)
+  const subscriptionRef = collection(
+    firestore,
+    `/users/${uid}/userNotificationFeed/`
+  )
+
+  const [isFilteringOrgs, setIsFilteringOrgs] = useState<boolean>(false)
+  const [isFilteringBills, setIsFilteringBills] = useState<boolean>(false)
+
+  const [allResults, setAllResults] = useState<Notifications>([])
+  const [orgResults, setOrgResults] = useState<Notifications>([])
+  const [billResults, setBillResults] = useState<Notifications>([])
+
+  const [notificationsDisplay, setNotificationsDisplay] =
+    useState<Notifications>([])
+  const [shouldDisplayResults, setShouldDisplayResults] =
+    useState<boolean>(true)
+
+  const onOrgFilterChange = (isFiltering: boolean) => {
+    const filterUpdatingTo = !isFiltering
+    setIsFilteringOrgs(filterUpdatingTo)
+
+    // isFilteringOrgs has to be inverted because setState (i.e. setIsFilteringOrgs)
+    // doesn't happen until a batch at the end
+
+    setShouldDisplayResults(true)
+    if (!isFilteringOrgs && isFilteringBills) {
+      return setShouldDisplayResults(false)
+    } else if (!isFilteringOrgs) {
+      return setNotificationsDisplay(billResults)
+    } else if (isFilteringBills) {
+      return setNotificationsDisplay(orgResults)
+    }
+    return setNotificationsDisplay(allResults)
+  }
+
+  const onBillFilterChange = (isFiltering: boolean) => {
+    const filterUpdatingTo = !isFiltering
+    setIsFilteringBills(filterUpdatingTo)
+
+    // isFilteringBills has to be inverted because setState (i.e. setIsFilteringBills)
+    // doesn't happen until a batch at the end
+
+    setShouldDisplayResults(true)
+    if (isFilteringOrgs && !isFilteringBills) {
+      return setShouldDisplayResults(false)
+    } else if (isFilteringOrgs) {
+      return setNotificationsDisplay(billResults)
+    } else if (!isFilteringBills) {
+      return setNotificationsDisplay(orgResults)
+    }
+    return setNotificationsDisplay(allResults)
+  }
+
+  const notificationQuery = async () => {
+    let notificationList: Notifications = []
+    const q = query(subscriptionRef, where("uid", "==", `${uid}`))
+    const querySnapshot = await getDocs(q)
+    querySnapshot.forEach(doc => {
+      // doc.data() is never undefined for query doc snapshots
+      notificationList.push(doc.data().notification)
+    })
+
+    if (notificationsDisplay.length === 0 && notificationList.length != 0) {
+      setAllResults(notificationList)
+      setOrgResults(
+        notificationList.filter(notification => notification.type === "org")
+      )
+      setBillResults(
+        notificationList.filter(notification => notification.type === "bill")
+      )
+      setNotificationsDisplay(allResults)
+    }
+  }
+
+  useEffect(() => {
+    uid ? notificationQuery() : null
+  })
```

Comment:
Currently this `useEffect` runs every render, which I don't think is what we want because it could introduce infinite loops.

We can introduce a dependency array to only fire the effect when any of the dependencies change. Since neither should change, then it should just query once. You can add a `console.count("query effect")` and check the console to ensure it only fires once.

```suggestion
  useEffect(() => {
    console.count("query effect")
    uid ? notificationQuery() : null
  }, [uid, notificationQuery])
```

Reply:
done